resolver_wrapper.go

```diff
@@ -78,7 +80,11 @@ func (ccr *ccResolverWrapper) start() error {
 			Authority:            ccr.cc.authority,
 		}
 		var err error
-		ccr.resolver, err = ccr.cc.resolverBuilder.Build(ccr.cc.parsedTarget, ccr, opts)
+		if ccr.cc.dopts.copts.UseProxy || ccr.cc.dopts.copts.Dialer != nil {
```

Comment:
I think you might have this condition wrong. See this line from the design:

This resolver is not used if WithNoProxy or WithCustomDialer is set -- instead, the normal name resolver determined by the scheme will be used directly.